coalib/core/Bear.py

```diff
@@ -0,0 +1,501 @@
+    ... 'Memory Leak', 'Documentation', 'Duplication', 'Commented Code',
+    ... 'Grammar', 'Missing Import', 'Unreachable Code', 'Undefined Element',
+    ... 'Code Simplification'}
+    >>> CAN_FIX = {'Syntax', ...}
+
+    Specifying something to ``CAN_FIX`` makes it obvious that it can be
+    detected too, so it may be omitted:
+
+    >>> class SomeBear(Bear):
+    ...     CAN_DETECT = {'Syntax', 'Security'}
+    ...     CAN_FIX = {'Redundancy'}
+    >>> sorted(SomeBear.can_detect)
+    ['Redundancy', 'Security', 'Syntax']
+
+    Every bear has a data directory which is unique to that particular bear:
+
+    >>> class SomeBear(Bear): pass
+    >>> class SomeOtherBear(Bear): pass
+    >>> SomeBear.data_dir == SomeOtherBear.data_dir
+    False
+
+    A bear can be dependant from other bears. ``BEAR_DEPS`` contains bear
+    classes that are to be executed before this bear gets executed. The results
+    of these bears will then be passed inside ``self.dependency_results`` as a
+    dict. The dict will have the name of the bear as key and a list of its
+    results as values:
+
+    >>> class SomeBear(Bear): pass
+    >>> class SomeOtherBear(Bear):
+    ...     BEAR_DEPS = {SomeBear}
+    >>> SomeOtherBear.BEAR_DEPS
+    {<class 'coalib.core.Bear.SomeBear'>}
+    """
+
+    LANGUAGES = set()
+    REQUIREMENTS = set()
+    AUTHORS = set()
+    AUTHORS_EMAILS = set()
+    MAINTAINERS = set()
+    MAINTAINERS_EMAILS = set()
+    PLATFORMS = {'any'}
+    LICENSE = ''
+    INCLUDE_LOCAL_FILES = set()
+    CAN_DETECT = set()
+    CAN_FIX = set()
+    ASCIINEMA_URL = ''
+    BEAR_DEPS = set()
+
+    @classproperty
+    def name(cls):
+        """
+        :return:
+            The name of the bear.
+        """
+        return cls.__name__
+
+    @classproperty
+    def can_detect(cls):
+        """
+        :return:
+            A set that contains everything a bear can detect, including
+            information from what it can fix too.
+        """
+        return cls.CAN_DETECT | cls.CAN_FIX
+
+    @classproperty
+    def source_location(cls):
+        """
+        Returns the directory this bear is inside.
+
+        >>> class SomeBear(Bear): pass
+        >>> SomeBear.source_location
+        '...Bear.py'
+        """
+        return inspect.getfile(cls)
+
+    @classproperty
+    def maintainers(cls):
+        """
+        :return:
+            A set containing ``MAINTAINERS`` if specified, else takes
+            ``AUTHORS`` by default.
+        """
+        return cls.AUTHORS if cls.MAINTAINERS == set() else cls.MAINTAINERS
+
+    @classproperty
+    def maintainers_emails(cls):
+        """
+        :return:
+            A set containing ``MAINTAINERS_EMAILS`` if specified, else takes
+            ``AUTHORS_EMAILS`` by default.
+        """
+        return (cls.AUTHORS_EMAILS
+                if cls.MAINTAINERS_EMAILS == set() else
+                cls.MAINTAINERS_EMAILS)
```

Comment:
otherwise you can't set them easily via `AUTHORS_EMAILS = ["a@b.com"]`.

Also we need to maintain compatibility